### DISCUSSION TITLE
[Feature] image editor, upscaler

### DISCUSSION THREAD

**Discussion by ilyaChuk2004** [immich-app/immich#10883] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be useful to have the functionality of a photo upscaler. I have old photos in poor quality. And I would like to make an improved version of them through the immich interface. Based on neural networks, of course. 
And while I was writing this, I noticed that there is not even a minimal photo editor. Some kind of cropping, color correction is needed.

I am a fullstack developer myself and would like to help create this functionality. But I don't understand where to start in this project?

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
An image editor is wanted, but fairly complicated - a few attempts at contributions in the past have failed. If you want to have a stab at it then that's certainly welcome, but please discuss your plans with us before getting started.

As for AI upscaling, I personally think that'd be a better fit for a plugin in the future.

**Comment by karmadelic702** [comment]
I would also like to see an AI upscaler feature. I have a lot of very old digital photos that I took on a 1MP camera about 26 years ago. It would be huge to have a feature like this for really old photos. Back in the day, those photos looked good, but on more modern hardware, they do not look as good anymore. I agree with the previous post that a feature like this should preserve the original photo.
